core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -792,10 +801,23 @@ class SmartConnectorRowRDD(_session: SnappySession,
   }
 
   override def computeResultSet(
-      thePart: Partition): (Connection, Statement, ResultSet) = {
+      thePart: Partition, context: TaskContext): (Connection, Statement, ResultSet) = {
     val helper = new SparkShellRDDHelper
     val conn: Connection = helper.getConnection(
       connProperties, thePart)
+    if (context ne null) {
+      val partitionId = context.partitionId()
+      context.addTaskCompletionListener { _ =>
+        logDebug(s"closed connection for task from listener $partitionId")
+        try {
+          conn.commit()
+          conn.close()
+          logDebug("closed connection for task " + context.partitionId())
+        } catch {
+          case NonFatal(e) => logWarning("Exception closing connection", e)
+        }
+      }
+    }
```

Comment:
@suranjan Oh, didn't notice this is the smart connector RDD. That will not be created on executor. However, in general the comment about relying only on context.addTaskCompletionListener for snapshot cleanup/commit remains. The ColumnInsertExec plan, for example, will be invoked on executor and cleanup if being done only via addTaskCompletionListener is not enough since context will be null for that case. It has to be done explicitly in ColumnBatchCreator in the two places it creates the ColumnInsertExec plan explicitly.